### DISCUSSION TITLE
Challenges chart not showing full node controller challenges

### DISCUSSION THREAD

**Discussion by hutchwilco** [guydavis/machinaris#604] [Q&A] [answered]
I'm still having a problem with the challenges chart - in the entire 6-8 months of getting to know Machinaris, I've had no challenges shown at all ( prior to 0.6.9 or 0.6.8 I would get a blank empty challenges chart - after those versions I just get no chart at all).

Recently I installed my first separate harvester - as soon as I got it working, the Challenges chart appeared on the main dashboard but only shows challenges for that separate harvester, it doesn't show any challenges for the fullnode/controller's harvester.

**Comment by guydavis** [comment, marked as answer]
Sorry to hear that, please try:

```
docker exec -it machinaris bash
cd /root/.chia/mainnet/log
grep --text -i eligible debug.log.1 debug.log
```
You should see output like:
![image](https://user-images.githubusercontent.com/12936/157352338-872a7f68-fc19-4e41-b23b-79f3dc86e7dc.png)

This is parsed and inserted into the Machinaris database for display.

To troubleshoot further, please provide:

1. A full screenshot of your Workers page.
2. A full screenshot of the Chia fullnode Worker page (clicked into that row from Workers page).
3. A full screenshot of your Summary page.
4. A full screenshot of your Index page (Machinaris link in top left)
5. A complete copy of your [apisrv.log](https://github.com/guydavis/machinaris/wiki/Troubleshooting#log-disk-locations)
6. A complete series of screenshots showing your Machinaris fullnode Unraid Docker configuration.  Or `docker-compose.yml` if using that instead.

Thanks!

**Reply by hutchwilco** [reply]
Thanks Guy! see info below.
The debug.log and ...log.1 output for 'eligible' show successful proofs found ( this is what's confusing to me about this issue. I haven't changed any ports from standard that I know of, timezones are all the same...).

<img width="1252" alt="Screen Shot 2022-03-10 at 8 25 45 AM" src="https://user-images.githubusercontent.com/4792731/157516622-fbb6d2f2-925f-4226-96dc-47dab3fe4d6e.png">

[docker-compose.txt](https://github.com/guydavis/machinaris/files/8217423/docker-compose.txt)
[apisrv.log](https://github.com/guydavis/machinaris/files/8217442/apisrv.log)
![10 0 1 96_8926_workers](https://user-images.githubusercontent.com/4792731/157515681-24dedd9a-7b39-4dd9-8184-4f26f10d5224.png)
![10 0 1 96_8926_worker_hostname=10 0 1 96 blockchain=chia](https://user-images.githubusercontent.com/4792731/157513991-ffa8a2e2-7462-45cd-8841-c7401aefbcf3.png)
![Screen Shot 2022-03-10 at 8 00 11 AM](https://user-images.githubusercontent.com/4792731/157516041-18b4bea4-1295-4ba4-b054-bde684807535.png)

![10 0 1 96_8926_index](https://user-images.githubusercontent.com/4792731/157515521-7670063b-e9be-484b-98ef-f001313b5a25.png)

**Reply by guydavis** [reply]
Thanks for all those details.  Please the following in your `docker-compose.yml` for the fullnode.  Change the `hostname: ` field to something shorter without hyphens.  Example:

```
version: '3.9'
services:
    machinaris:
        image: ghcr.io/guydavis/machinaris
        container_name: machinaris
        hostname: chiafullnode1
        restart: always
...
```

No need to change the actual hostname of the machine itself, just the `hostname` value passed into the container when Docker launches.

Then after 15 minutes once restarted, go to the Workers page, and select the stale row with hostname `gridshare-ONDA-Chia-D32H-D4-V1-01`.  Select that stale row and click 'Prune Selected'. 

After a bit longer, check your index page again. Hope this helps!

**Reply by hutchwilco** [reply]
🤦🏻‍♂️ It's always the simplest, dumbest answer, hiding in plain sight isn't it!
I never liked that silly long hostname anyway - someone else set it and I never got round to updating it!

It's worth noting that I didn't have to Prune the old worker (I did remove the container before restarting - not sure if that was what you intended) - anyway, it has automagically updated the worker hostname and the graph is now displaying both harvester and fullnode.
Thanks!
